additional support mmdet v3

wkpark opened the pull request:
minimal fix applied for the mmdet v3 
* [x] mmdet v3 support added based on Bing-su's work 21bf8952, deeb4ac7f, c2f410f6d
   * [x] mmdet v3 fixes 
   * [x] config files for mmdet v3
 
Dependency issues could be a challenge for end users, and supporting both versions could be a burden for developers; however, I believe it would be good to offer support for both versions in the meantime.

- [x] check mmdet_dd-person_mask2former-v3.py and the new model_data for it. **DONE**
  - https://github.com/dustysys/ddetailer/issues/5#issuecomment-1311231989
  - https://github.com/open-mmlab/mmdetection/tree/main/configs/mask2former#instance-segmentation
    - [x] segm model file : https://download.openmmlab.com/mmdetection/v3.0/mask2former/mask2former_r50_8xb2-lsj-50e_coco/mask2former_r50_8xb2-lsj-50e_coco_20220506_191028-41b088b6.pth

## HOW TO USE

- delete `extensions/ddetailer`
- delete `models/mmdet`
- install extension `https://github.com/wkpark/ddetailer` with `mmdet-v3` branch
- press install button and restart webui!
![image](https://github.com/dustysys/ddetailer/assets/232347/a349c7ad-a32d-4b00-9f55-0c86d64d9c32)

wkpark committed:
add configs from original repo https://huggingface.co/dustysys/ddetailer/blob/main/mmdet

wkpark committed:
support partial update mmdet configs

wkpark committed:
support mmdet v3 based on Bing-su's work

wkpark committed:
add config files  for mmdetection v3

wkpark committed:
update for mmdet v3

wkpark committed:
check the pytorch version and conditially install mmdet

svrakiss commented on scripts/ddetailer.py:
```diff
@@ -44,11 +44,17 @@ def modeltitle(path, shorthash):
+            run(f'"{python}" -m pip install mmdet==2.28.2', desc=f"Installing mmdet", errdesc=f"Couldn't install mmdet")
+        else:
+            run(f'"{python}" -m mim install mmcv>==2.0.0', desc=f"Installing mmcv", errdesc=f"Couldn't install mmcv")
+            run(f'"{python}" -m pip install mmdet', desc=f"Installing mmdet", errdesc=f"Couldn't install mmdet")
```
You should specify mmdet>=3 so it is more readable

wkpark committed:
specify the mmdet version

wkpark committed:
update num_things_classes=80 for the new model file

wkpark committed:
download the segm model file from openmmlab.com for mmdet v3

Please see https://github.com/open-mmlab/mmdetection/tree/main/configs/mask2former#instance-segmentation
for R-50 backbone

svrakiss commented on scripts/ddetailer.py:
```diff
@@ -61,7 +61,13 @@ def startup():
+            load_file_from_url(
+                "https://download.openmmlab.com/mmdetection/v3.0/mask2former/mask2former_r50_8xb2-lsj-50e_coco/mask2former_r50_8xb2-lsj-50e_coco_20220506_191028-41b088b6.pth",
+                segm_path,
+                file_name="mmdet_dd-person_mask2former.pth")
```
This is fine for fresh installs but not updates. This would not replace the previous model file when upgrading from mmdet<3. More importantly, nothing seems to be going wrong (no exceptions or warnings are raised) when using the v3 config with the old model other than never detecting anything.

wkpark replied:
good point. it needs to consider cases where the models and config don't match etc.
Simply removing the `models/mmdet` dir will suffice for developers, but for end users more sophisticated method is required such as adding a stamp file and verifying it internally.

Advtech92 commented:
It looks like the plugin is broken again as it can't download the mmdet.core again
